tests/model_explainability/nemo_guardrails/utils.py

```
@@ -0,0 +1,304 @@
+        host: Hostname for NeMo Guardrails
+        token: Authentication token (None if auth disabled)
+        ca_bundle_file: Path to CA bundle file for TLS verification
+        health_endpoint: Health check endpoint path (default: "/")
+
+    Raises:
+        AssertionError: If health check fails after retries
+    """
+    url = f"https://{host}{health_endpoint}"
+    headers = get_auth_headers(token=token) if token else {}
+
+    response = requests.get(url=url, headers=headers, verify=ca_bundle_file)
```

Ruff (0.15.9)

[error] Probable use of `requests` call without timeout

(S113)